broker/src/main/java/io/zeebe/broker/clustering/atomix/AtomixFactory.java

```diff
@@ -97,13 +98,15 @@ private static RaftPartitionGroup createRaftPartitionGroup(
                 (raftContext, threadContext, threadContextFactory) ->
                     new ZeebeRaftStateMachine(raftContext))
             .withSnapshotStoreFactory(new DbSnapshotStoreFactory())
+            .withStorageLevel(dataCfg.getAtomixStorageLevel())
             .withFlushOnCommit();
 
     // by default, the Atomix max entry size is 1 MB
     final int maxMessageSize = (int) networkCfg.getMaxMessageSizeInBytes();
     partitionGroupBuilder.withMaxEntrySize(maxMessageSize);
 
-    Optional.ofNullable(dataCfg.getLogSegmentSizeInBytes())
+    Optional.ofNullable(dataCfg.getLogSegmentSize())
```

Comment:
Not sure this change makes things better as you now have a dependency to Spring DataSize in this class

Reply:
Hadn't thought of that; I just corrected it as afaik `Optional.ofNullable` of a primitive is never null?

Reply:
true, this is somewhat misleading. getLogmentSize will never be null.

Reply:
I moved the `Optional.ofNullable` to the configuration class, let me know what you think.